src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -220,6 +220,21 @@ ApplicationSharedPtr FindApp(DataAccessor<ApplicationSet> accessor,
 }
 
 template <class UnaryPredicate>
+ApplicationSharedPtr FindApp(
+    DataAccessor<ApplicationManagerImpl::AppsWaitRegistrationSet> accessor,
+    UnaryPredicate finder) {
+  ApplicationManagerImpl::AppsWaitRegistrationSet::iterator it = std::find_if(
```

Comment:
@AGritsevich looks ugly. Isn't it better to request appropriate application using the mutex and avoid this `DataAccessor`?